modules/fundamental/src/vulnerability/service/mod.rs

```diff
@@ -155,6 +162,141 @@ impl VulnerabilityService {
 
         Ok(result.rows_affected)
     }
+
+    pub async fn analyze_purls<C>(
+        &self,
+        purls: Vec<&str>,
+        connection: &C,
+    ) -> Result<HashMap<String, Vec<VulnerabilityDetails>>, Error>
+    where
+        C: ConnectionTrait + StreamTrait,
+    {
+        if purls.is_empty() {
+            return Ok(Default::default());
+        }
+
```

Comment:
You shouldn't need this. We used it before because both the `by_uuid` and `by_purl` fn's were always being called, whether they were needed or not.

In this case, our query will be "" if there aren't any purl's, and sea_orm should be smart enough to know it doesn't need to execute an empty string. :smile: